oxide-vpc/tests/integration_tests.rs

```diff
@@ -2024,3 +2046,369 @@ fn gateway_icmpv6_ping() {
         _ => panic!("expected Echo Reply, got {:?}", reply_icmp),
     }
 }
+
+// Generate a packet containing an NDP Router Solicitation.
+//
+// The source MAC is used to generate the source IPv6 address, using the EUI-64
+// transform. The resulting packet is has a multicast MAC address, and the
+// All-Routers destination IPv6 address.
+fn gen_router_solicitation(src_mac: &MacAddr) -> Packet<Parsed> {
+    use smoltcp::wire::Icmpv6Packet;
+    use smoltcp::wire::Icmpv6Repr;
+    use smoltcp::wire::IpAddress;
+    use smoltcp::wire::NdiscRepr;
+    use smoltcp::wire::RawHardwareAddress;
+
+    // The source IPv6 address is the EUI-64 transform of the source MAC
+    let src_ip = Ipv6Addr::from_eui64(src_mac);
+
+    // Must be destined for the All-Routers IPv6 address, and the corresponding
+    // multicast Ethernet address
+    let dst_ip: Ipv6Addr = "ff02::2".parse().unwrap();
+    let dst_mac = MacAddr::from_const([0x33, 0x33, 0x00, 0x00, 0x00, 0x02]);
+
+    let solicit = NdiscRepr::RouterSolicit {
+        lladdr: Some(RawHardwareAddress::from_bytes(&src_mac.bytes())),
+    };
+    let req = Icmpv6Repr::Ndisc(solicit);
+    let mut body_bytes = vec![0u8; req.buffer_len()];
+    let mut req_pkt = Icmpv6Packet::new_unchecked(&mut body_bytes);
+    let mut csum = CsumCapab::ignored();
+    csum.icmpv6 = smoltcp::phy::Checksum::Tx;
+    let _ = req.emit(
+        &IpAddress::Ipv6(src_ip.into()),
+        &IpAddress::Ipv6(dst_ip.into()),
+        &mut req_pkt,
+        &csum,
+    );
+    let mut ip6 = Ipv6Hdr::from(&Ipv6Meta {
+        src: src_ip,
+        dst: dst_ip,
+        proto: Protocol::ICMPv6,
+    });
+    ip6.set_total_len(ip6.hdr_len() as u16 + req.buffer_len() as u16);
+    let eth = EtherHdr::from(&EtherMeta {
+        dst: dst_mac,
+        src: *src_mac,
+        ether_type: ETHER_TYPE_IPV6,
+    });
+
+    let mut pkt_bytes =
+        Vec::with_capacity(ETHER_HDR_SZ + ip6.hdr_len() + req.buffer_len());
+    pkt_bytes.extend_from_slice(&eth.as_bytes());
+    pkt_bytes.extend_from_slice(&ip6.as_bytes());
+    pkt_bytes.extend_from_slice(&body_bytes);
+    Packet::copy(&pkt_bytes).parse().unwrap()
+}
+
+// Verify that a Router Solicitation emitted from the guest results in a Router
+// Advertisement from the gateway.
+#[test]
+fn gateway_router_advert_reply() {
+    use smoltcp::time::Duration;
+    use smoltcp::wire::Icmpv6Packet;
+    use smoltcp::wire::Icmpv6Repr;
+    use smoltcp::wire::Ipv6Address;
+    use smoltcp::wire::NdiscRepr;
+    use smoltcp::wire::NdiscRouterFlags;
+    use smoltcp::wire::RawHardwareAddress;
```

Comment:
I know I've done this myself in some places, but we might as well move these imports to module level.

Reply:
Sure, we can do that. I'm kinda playing fast-and-loose on the assumption we'll do a bulk format at some point.

Reply:
Yes, once we both have a pause point let's do bulk reformat. In this case I was just saying that the imports are used in more than one function, so might as well do it at module level.

Reply:
Ohhh, I interpreted that as the "module-level" grouping, the option to `rustfmt` that groups imports at the level of the module. You mean put the imports in the integration test module itself, rather than inside the function. My bad, yeah I get what you're saying now. I'll move them up.